modules/ztp-image-based-upgrade-procedure-steps.adoc

```diff
@@ -0,0 +1,191 @@
+====
+
+.Prerequisites
+
+* Log in to the hub cluster as a user with `cluster-admin` privileges.
+* Create policies and `ConfigMap` objects for resources used in the image-based upgrade.
+* Install {lcao} and OADP Operators on all managed clusters through the hub cluster.
+
+.Procedure
+
+. Create a YAML file on the hub cluster that contains the `ImageBasedGroupUpgrade` CR, for example, named `example-group-upgrade.yaml`:
++
+--
+[source,yaml]
+----
+apiVersion: lcm.openshift.io/v1alpha1
+kind: ImageBasedGroupUpgrade
+metadata:
+  name: example-group-upgrade
+  namespace: default
+spec:
+  clusterLabelSelectors: <1>
+  - matchExpressions:
+    - key: name
+      operator: In
+      values:
+      - spoke1
+      - spoke4
+      - spoke6
+  ibuSpec:
+    seedImageRef: <2>
+      image: quay.io/seed/image:4.16.0-rc.1
+      version: 4.16.0-rc.1
+      pullSecretRef:
+        name: "<seed_pull_secret>"
+  extraManifests: <3>
+    - name: example-extra-manifests
+      namespace: openshift-lifecycle-agent
+  oadpContent: <4>
+    - name: oadp-cm
+      namespace: openshift-adp
+  plan: <5>
+  - actions: ["Prep"]
+    rolloutStrategy:
+      maxConcurrency: 2
+      timeout: 2400
+----
+<1> Specify the set of clusters that you want to upgrade.
+<2> Defines the target platform version, the seed image to be used, and the secret required to access the image.
+<3> (Optional) Specify the list of `ConfigMap` resources that contain your custom catalog sources to retain after the upgrade, and your extra manifests to apply to the target cluster that are not part of the seed image.
+<4> Specify the list of `ConfigMap` resources that contain the OADP `Backup` and `Restore` CRs.
+<5> Defines the upgrade plan.
+--
+
+. Apply the `example-group-upgrade.yaml` file by running the following command on the hub cluster:
++
+[source,terminal]
+----
+$ oc apply -f example-group-upgrade.yaml
+----
+
+. Monitor the status updates by running the following command on the hub cluster:
++
+--
+[source,terminal]
+----
+$ oc get ibgu -o yaml
+----
+
+.Example output
+[source,yaml]
+----
+[...]
+status:
+  clusters:
+  - completedActions:
+    - action: Prep
+    name: spoke1
+  - completedActions:
+    - action: Prep
+    name: spoke4
+  - failedActions:
+    - action: Prep
+    name: spoke6
+[...]
+----
+
+This example plan starts with the `Prep` stage only and you append actions to the plan based on the results of the step
+{cgu-operator} adds a label to the clusters to mark if the upgrade succeeded or failed.
+For example, the `lcm.openshift.io/ibgu-prep-failed` is applied to clusters that failed the `Prep` stage.
+
+After investigating the reason for failure, you can append the `AbortOnFailure` step to move the clusters labeled with `lcm.openshift.io/ibgu-<action>-failed` back to the `Idle` stage.
+This deletes any resources on the selected clusters that are related to the upgrade and allows you to retry the upgrade after troubleshooting.
```

Comment:
`This deletes any resources...`--this is not clear to me. What exactly (running a command or a setting a setting) deletes those resources?

Reply:
"Allows you" is a form of anthrophomorphism.

> This deletes any resources on the selected clusters that are related to the upgrade so that you can retry the upgrade after troubleshooting.

Reply:
+1 to Max about the "delete" action.

Reply:
Setting back to `Idle` is what triggers an Operator to delete the resources. [Context here](https://80111--ocpdocs-pr.netlify.app/openshift-enterprise/latest/edge_computing/image_based_upgrade/cnf-understanding-image-based-upgrade#cnf-image-based-upgrade-concept-idle_understanding-image-based-upgrade)